blog(embertimes161): template-colocation-migrator (#746)
ember-learn/ember-blog: source/2020-08-21-the-ember-times-issue-161.md

<SAYING-HELLO-IN-YOUR-FAVORITE-LANGUAGE> Emberistas! 🐹

<SOME-INTRO-HERE-TO-KEEP-THEM-SUBSCRIBERS-READING>
ember-component-template-colocation-migrator now supports addons 🎉,
READMORE

---

## [Release of ember-component-template-colocation-migrator 1.1.0 🎉](https://github.com/ember-codemods/ember-component-template-colocation-migrator/releases/tag/v1.1.0)

> The [ember-component-template-colocation-migrator](https://github.com/ember-codemods/ember-component-template-colocation-migrator) codemod will co-locate component JS and HBS files as described in the [Component Templates Co-location RFC](https://emberjs.github.io/rfcs/0481-component-templates-co-location.html). The v1.1.0 release adds [support for migrating addons](https://github.com/ember-codemods/ember-component-template-colocation-migrator/pull/14) although templates shared by multiple components are excluded. In this case, it is up to end users how to list those; they are reported in the same way other skipped templates (like partials) are. Similarly Mixins which use layout are not migrated.

Reviewer: I came up with a few suggestions:

- "will co-locate" -> "co-locates"
- "although" -> "but"
- "Mixins which" -> "mixins that"

```suggestion
The [ember-component-template-colocation-migrator](https://github.com/ember-codemods/ember-component-template-colocation-migrator) codemod co-locates component JS and HBS files as described in the [Component Templates Co-location RFC](https://emberjs.github.io/rfcs/0481-component-templates-co-location.html). The v1.1.0 release adds [support for migrating addons](https://github.com/ember-codemods/ember-component-template-colocation-migrator/pull/14), but templates shared by multiple components are excluded. In this case, it is up to end users how to list those; they are reported in the same way other skipped templates (like partials) are. Similarly, mixins that use layout are not migrated.
```

When running the codemod, instead of:

```sh
your-addon/
  addon/
    components/
      some-component.js
    templates/
      components/
        some-component.hbs
```

You will get:

```sh
your-addon/
  addon/
    components/
      some-component.js
      some-component.hbs
```

> Try it out today by running the migrator in your app: `npx github:ember-codemods/ember-component-template-colocation-migrator`

Reviewer: ```suggestion
Try it out today by running the migrator in your app or addon: `npx github:ember-codemods/ember-component-template-colocation-migrator`
```
